### ISSUE TITLE
Variable mappings are serialized to save files

### ISSUE THREAD

**Issue by tellowkrinkle** [07th-mod/higurashi-assembly#25]
For some reason the game includes the mapping of variable name to number in its save format.  This means that if we add new flags (like `LConsoleArc` which was causing issues for me), they'll break old save files since the old save files will update the mapping dictionary to not have those flags

Does anyone know of any reason this would be useful?  If not, we should ignore it when we load saves so that our old save compatibility is at least slightly better.

https://github.com/07th-mod/higurashi-assembly/blob/f7804b3acbf683eee9c5b2d417a13ac8f956afa2/Assets.Scripts.Core.Buriko/BurikoMemory.cs#L370

https://github.com/07th-mod/higurashi-assembly/blob/f7804b3acbf683eee9c5b2d417a13ac8f956afa2/Assets.Scripts.Core.Buriko/BurikoMemory.cs#L415

**Comment by idealpersona** [comment]
Just saw this.  Since the number mappings are hard-coded it may not be needed.  However we did change the way this portion of the code works a bit to also support adding new flags to an existing game, if I remember correctly - you can see through the history or a mg->mod diff on BurikoMemory.  Let me know if I should track it down - could be related, but in either case lets be cautious about this one.  Not opposed to changing it though, if it helps the save compatibility.

**Comment by tellowkrinkle** [comment]
Right, that change was to make it so the values of those variables are loaded without overwriting the defaults.  This is loading the string to int variable name definitions.  In fact the way this currently works means that the other fancy variable loading code will never help, since any save file that would have used it will undefine the existence of the new variables on load
